File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        count = defaultdict(list)
        res = []
        for s in strs:
            alphboard = [0] * 26
            for i in range(len(s)):
                alphboard[ord(s[i]) - ord('a')] += 1
            count[tuple(alphboard)].append(s)
        for lst in count.values():
            res.append(lst)
        return res

File: 0049-group-anagrams/0049-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> res = new HashMap<>();
        for(String s: strs){
            int[] board = new int[26];
            for (char c: s.toCharArray()){
                board[c - 'a'] ++;
            }
            String key = Arrays.toString(board);
            res.putIfAbsent(key, new ArrayList<>());
            res.get(key).add(s);
        }
        return new ArrayList<>(res.values());
    }
}